src/containers/org/people/PeopleTable.js

```diff
@@ -0,0 +1,294 @@
+import memberHasSelectedRoles from './utils/memberHasSelectedRoles';
+
+import AddMembersToOrganizationModal from '../components/AddMembersToOrganizationModal';
+import AssignRolesToMembersModal from '../components/AssignRolesToMembersModal';
+import { selectCurrentRoleAuthorizations } from '../../../core/redux/selectors';
+import { getUserProfile } from '../../../utils';
+import {
+  FILTER,
+  INITIAL_PAGINATION_STATE,
+  PAGE,
+  paginationReducer
+} from '../../../utils/stateReducers/pagination';
+import { RemoveMemberFromOrgModal, RemoveRoleFromMemberModal } from '../components';
+import { filterOrganizationMember } from '../utils';
+
+const { NEUTRAL } = Colors;
+
+const TableToolbar = styled.div`
+  align-items: center;
+  display: grid;
+  grid-gap: 8px;
+  grid-template-columns: minmax(160px, max-content) 160px 1fr 100px 160px;
+  margin: 16px 0;
+`;
+
+const MembersCheckboxWrapper = styled.div`
+  margin: 0 8px 0 17px;
+`;
+
+const Table = styled.table`
+  border: 1px solid ${NEUTRAL.N100};
+  border-collapse: collapse;
+  width: 100%;
+`;
+
+const Selection = styled.span`
+  margin-left: 10px;
+  vertical-align: middle;
+`;
+
+const PlusIcon = <FontAwesomeIcon icon={faPlus} size="lg" />;
+
+const MAX_PER_PAGE = 20;
+
+type Props = {
+  isOwner :boolean;
+  members :List;
+  organizationId :UUID;
+  roles :Role[];
+};
+
+const PeopleTable = ({
+  isOwner,
+  members,
+  organizationId,
+  roles,
+} :Props) => {
+
+  const currentRoleAuthorizations :Map = useSelector(selectCurrentRoleAuthorizations());
+  // consider using reducers for handling member/role/action selection
+  const [isVisibleRemoveMemberFromOrgModal, setIsVisibleRemoveMemberFromOrgModal] = useState(false);
+  const [isVisibleRemoveRoleFromMemberModal, setIsVisibleRemoveRoleFromMemberModal] = useState(false);
+  const [isVisibleAddMembersToOrganizationModal, setIsVisibleAddMembersToOrganizationModal] = useState(false);
+  const [isVisibleAssignRolesModal, setIsVisibleAssignRolesModal] = useState(false);
+  const [paginationState, paginationDispatch] = useReducer(paginationReducer, INITIAL_PAGINATION_STATE);
+  const [targetMember, setTargetMember] = useState();
+  const [targetRole, setTargetRole] = useState();
+  const [selectedFilters, setSelectedFilters] = useState(Map({
+    role: Set(),
+    authorization: Set(),
+  }));
+  const [selectedMembers, setSelectedMembers] = useState(Map());
+
+  let filteredMembers = members;
+  // filter by query
+  if (paginationState.query) {
+    filteredMembers = filteredMembers.filter((member) => filterOrganizationMember(member, paginationState.query));
+  }
+  // filter by selected roles
+  filteredMembers = filteredMembers
+    .filter((member) => memberHasSelectedRoles(member, selectedFilters.get('role', Set())));
+  // filter by selected authorization
+  filteredMembers = filteredMembers
+    .filter((member) => memberHasSelectedIdentityTypes(member, selectedFilters.get('authorization', Set())));
```

Comment:
you could chain these filters if you want, but maybe you want to have those comments interspersed